[11.0][MIG] l10n_es_aeat_mod347: Migration to 11.0

angelmoya opened the pull request:
Migración + mejora del modelo 347 a v11.

cc @Tecnativa

Commit:
[REF] *: Refactorización para portar a 6.0 - Paso 1.

- Se han renombrado los módulos para usar la nomenclatura propuesta
  por OpenERP: l10n_es para el módulo base de localización (plan de
  cuentas), l10n_es_* para el resto de módulos.

- Se eliminan los módulos extra_addons/* que deberían moverse a
  los extra-addons genéricos (no son específicos de España).

- Se renombran los __terp__.py por __openerp__.py

Commit:
[REF][REN][FIX] Corregimos errores en los nombres de dependencias de ciertos módulos, por los correspondientes que se modificaron para esta versión 6.0 y en ciertos __init__ adaptamos los imports a los nuevos nombres de los ficheros. Renombrado de los archivos de traducción españoles de es_ES.po a es.po y pequeñas refactorizaciones

Commit:
[FIX+IMP] l10n_es_aeat_mod347: 2 things

* FIX: Renombrado de los archivos de traducción catalanas de ca_ES.po a ca.po
* IMP: Añadidos avisos NO ADAPTADO TODAVÍA A LA VERSIÓN 6.0 a varios módulos, limpieza l10n_es_partner_mercantil

Commit:
[ADD] l10n_es_aeat, l10n_es_aeat_347, l10n_es_aeat_349: Adds a generic module for aeat models, 347 module was portedto v6.0 and adds new module to print AEAT model 349.

Commit:
[IMP+FIX] l10n_es_aeat, l10n_es_aeat_mod347, l10n_es_aeat_mod349: Pequeñas mejoras generales (vistas, traducciones, código) y corrección pequeño bug por un olvido en la adaptación de la v5 a la v6

Commit:
[IMP][FIX] Nos traemos los cambios hechos ayer sobre la versión 5 del módulo del 347, que incluían los cambios hechos por la AEAT para la declaración 2010, para la versión 6.0 del módulo, además corregimos un bug encontrado por Jordi en la versión 5.0 hoy. También añado una nueva comprobación de los registros de empresas, ya que el cif de la empresa es requerido, por lo que no va a dejar confirmar el informe, mientras no se rellene este campo en todos los registros de empresa. La agrupación de pagos en efectivo se dejó igual que en la 5.0, ya que se sigue comportando como en la 5.0, sigo animando a que si alguien conoce como mejorarla adelante.

Commit:
[IMP] l10n_es_aeat_mod347: Actualizada traducción catalana después de las mejoras introducidas en el anterior commit

Commit:
[REF] Cambio de licencia de GPL3 a AGPL en los módulos donde sólo aparecía Pexego en la licencia, quedan los compartidos.

albertca committed:
l10n_es_aeat_mod347: Actualizado el módulo para realizar el nuevo modelo con el detalle de los trimestes obligatorio para la declaración del 2011.

Commit:
[ADD][REF] l10n_ES_aeat_mod347: Añadimos la posibilidad de agrupar por Cif/Nif en el cálculo del 347, por si hay más de una empresa con el mismo cif..., pequeña refactorización usando rec_name en lugar de name_get

Commit:
[FIX] l10n_es_aeat_mod347: Corregimos la agrupación por cif

Commit:
[IMP] l10n_es_aeat_mod347: Actualizadas plantillas y traducciones con los nuevos strings después de los recientes cambios en el 347.

acysos committed:
[FIX] l10n_es_aeat_mod347: Exclusión del IRPF del total de las facturas. Necesario para las Sociedades Irregulares, Sociedades Civiles y Comunidad de Bienes sin regimen de módulos en el IRPF. Al resto de las empresas no les afecta ya que no facturan con IRPF. l10n_es_aeat_mod340: Correción de los autores en el modelo 340

Commit:
Commit inicial de una rama para la versión 7.0. Todos los módulos fueron marcados como no instalables mientras no se compruebe que funcionan o migren, para poder usarlos o probarlos se debería poner el atributo installable de los ficheros __openerp__.py de cada módulo a True

Commit:
Launchpad automatic translations update.

Commit:
[FIX] Fix some typos in pot and po files.

Commit:
Launchpad automatic translations update.

percevaq committed:
[FIX]: Update view definitions to version 7.0.
[FIX]: Add security definition for objects l10n_es_aeat_mod347_calculate_records and l10n_es_aeat_mod347_export_to_boe.
[FIX]: Changed types to orm.Model, orm.TransientModel and orm.AbstractModel.
[FIX]: Contributions have been written in the standard format of the community.
[FIX]: Remove 'init_xml' keys, because it's no longer needed in v7.
[FIX]: Remove __author__ variables in files, because authors are put on manifest file (__openerp__.py).
[FIX]: Increased compatibility with standard PEP8.
[FIX]: Imports in standard format.
[IMP]: Add "state" field in tree view.
[IMP]: Option to generate report suppliers only.
[FIX]: Update address partner search to V7.
[FIX]: Update state in address partner to V7.
[FIX]: Update dependencies, now required the module documents to view generated file in V7.
[FIX]: Translate .pot file to v7.
[FIX]: Translate field "only_supplier" to spanish language.
[FIX]: Format the text that explains the module.
[IMP]: Add icon and screencapture.

Commit:
Launchpad automatic translations update.

Commit:
[IMP] l10n_es_aeat_mod347:
- Cambiado flag de los partners para no incluirlos en el 347 (opción menos común), que incluirlos (que suelen ser todos).
- Eliminado flag para incluir los diarios, ya que no se activa por defecto y el usuario se confunde con los resultados. Se cogen todos los diarios de caja.
- Otras mejoras menores.

Commit:
Launchpad automatic translations update.

Commit:
[IMP] l10n_es_aeat_mod347: Comentarios eliminados y alguna mejora de docstrings.

acysos committed:
[FIX] l10n_es_aeat_mod347: eliminación de las facturas canceladas en el calculo del 347

Commit:
Launchpad automatic translations update.

Commit:
[FIX] l10n_es_aeat_mod347: Corregido cálculo del importe para facturas rectificativas.

Commit:
[IMP] Eliminación de la mayoría de warnings que da runbot.

Commit:
Launchpad automatic translations update.

pedrobaeza committed:
[MIG] Mergear cambios de la 7 y recolocar

pedrobaeza committed:
[MIG] l10n_es_aeat_mod347: Migrado a nueva API

Commit:
[IMP] unificar campos en el módulo base + migrar a la nueva api.

cubells committed:
[IMP] Mejora para poder eliminar facturas selectivamente del modelo 347

pedrobaeza committed:
[FIX+IMP] l10n_es_aeat_mod347: Mejoras en la vista de facturas + Autores/patrocionadores expandidos + PEP8

oca-transbot committed:
OCA Transbot updated translations from Transifex

Commit:
[FIX] l10n_es_aeat_mod347, 340: Fixed terms for 'real estate' in fields, variables, classes and translations, and related PEP8 fixes.

pedrobaeza committed:
[FIX] l10n_es_aeat_mod347: copy no existe para las listas

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[FIX] l10n_es_aeat_mod347: Compatibilidad API vieja y nueva en la creación de periodos mensuales. Fixes #209

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[IMP] l10n_es_aeat_mod347: Eliminar la selección de periodo para no confundir

oca-transbot committed:
OCA Transbot updated translations from Transifex

jalzaga committed:
fix code errors in received cash amount

pedrobaeza committed:
[FIX+IMP] l10n_es_aeat_mod347: 3 things:

* FIX: Avoid error when multiple partners with the same VAT
* FIX: Fix onchange_partner_id of partner records
* IMP: Allow to select periods for the calculation

omar7r committed:
[FIX] l10n_es_aeat_mod347: Se corrige una incorrecta selección de empresas sobre las que se calculaba el informe.

oca-transbot committed:
OCA Transbot updated translations from Transifex

angelmoya committed:
WIP MIG mod347

antespi committed:
[8.0][IMP][l10n_es_aeat_mod347] Improve performance and other things (#317)

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[IMP] l10n_es_aeat_mod347: General code check

angelmoya commented:
@pedrobaeza he revisado lo de modificar la forma de obtener los datos de las facturas para obtenerlo de los apuntes, pero no lo tengo del todo claro. 

¿Sería correcto sacar los apuntes de los diarios que sean de compras o ventas en lugar de sacarlas de facturas?

No estoy seguro si es correcto que actualmente está utilizando la fecha de factura, si lo hacemos desde apuntes estaríamos utilizando la fecha contable.

He empezado con la parte de crear la plantilla de email y añadir un estado a cada registro de empresa. Me faltaría hacer una acción masiva para enviar el email mas de un cliente a la vez.

¿Sería correcto enviar esta información a todos los partners, o únicamente se enviaría a los españoles?

Tengo pensado hacer una acción para seleccionar desde el listado y enviar a todos los seleccionados, pero quizás sería bueno el envío con un botón sin tener que filtrar.

oihane commented on l10n_es_aeat_mod347/__manifest__.py:
```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
```
The first line # -*- coding: utf-8 -*- is not needed anymore in Python 3 if you save it in that encoding. (You can use the following command to auto-delete it: find . -name "*.py" -exec sed -i "/#.*coding\: /d" {} \;)

oihane commented:
PEP8 errors
```
l10n_es_aeat_mod347/models/mod347.py:749:80: E501 line too long (86 > 79 characters)
l10n_es_aeat_mod347/wizards/l10n_es_aeat_mod347_send_mail.py:3:1: E302 expected 2 blank lines, found 1
```

angelmoya commented:
Gracias Oiane, pero esta WIP... Cuando lo tenga aviso.

Si quieres ir probando funcionalidad:
- se ha añadido informe pdf para el detalle de cada registro de empresa.
- se ha añadido plantilla de email con ese informe para enviar detalle a
empresas
- se ha añadido estado y botones para gestionar los estados de cada registro
- wip añadir asistente para el envío masivo de estos emails desde el
listado de registro de empresas

El mié., 13 jun. 2018 11:36, oihane <notifications@github.com> escribió:

> PEP8 errors
>
> l10n_es_aeat_mod347/models/mod347.py:749:80: E501 line too long (86 > 79 characters)
> l10n_es_aeat_mod347/wizards/l10n_es_aeat_mod347_send_mail.py:3:1: E302 expected 2 blank lines, found 1
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/OCA/l10n-spain/pull/862#issuecomment-396876916>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AGcAhbkVJ_GFCwk37xhhjqSlhjvMzi1iks5t8N0TgaJpZM4UiVG_>
> .
>

Joaco1980 commented:
si finalmente no se cogen los datos de los apuntes se volverian a quedar fuera las facturas simplificadas del POS para el calculo del 347, se puede dar el caso de empresas que vendan por ambos canales, hasta donde yo se es perfectamente valido vender a un profesional a traves del POS siempre y cuando no supere la operacion 400 € IVA incl.

angelmoya commented:
Gracias por el aporte Joaquín, estoy en proceso y lo haremos a partir de
apuntes.

El dom., 17 jun. 2018 22:58, Joaquin Gutierrez <notifications@github.com>
escribió:

> si finalmente no se cogen los datos de los apuntes se volverian a quedar
> fuera las facturas simplificadas del POS para el calculo del 347, se puede
> dar el caso de empresas que vendan por ambos canales, hasta donde yo se es
> perfectamente valido vender a un profesional a traves del POS siempre y
> cuando no supere la operacion 400 € IVA incl.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/OCA/l10n-spain/pull/862#issuecomment-397906159>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AGcAhe9rlfhuw4KjW4vL8VXrIeJ3mQp6ks5t9sMGgaJpZM4UiVG_>
> .
>

fgarcia-humanoide commented:
Si se queda fuera lo del pos no sería un gran inconveniente.

Todo el mundo asocia 347 con facturas de verdad, no se va a esperar que se tenga en cuenta el pos tambíen, eso es rizar el rizo.

(en mi humilde opinion)

Joaco1980 commented:
Entonces que hacemos con los profesionales que compran en una tienda y les das una factura simplificada perfectamente deducible ya que tiene sus datos fiscales, el desglose de los impuestos y no supera los 400€ IVA incl.  Porque el importe total de operaciones que es lo que presentas con el 347 también debería de sumar ese tipo de operaciones....

Eso de que todo el mundo asocia el 347 con facturas de verdad me deja asombrado, otra cosa es que tus clientes no tengan determinada casuística pero con la ley en la mano las facturas simplificadas son perfectamente deducibles siempre y cuando tengan los datos fiscales, es mas te dire que ahora en el SII a Hacienda le tienes que diferenciar entre Simplificadas que se vayan a deducir (con datos fiscales) y Simplificadas que no se vayan a deducir porque no constan esos datos.

Yo opto porque cuando se realice una cosa de estas incluyan el mayor numero de casos posibles, yo no tengo ni idea de programar en Python pero supongo que buscar en los asientos contables de las cuenta 430 el DEBE-HABER y en la 400 y 410 el HABER-DEBE no creo que suponga gran cantidad de lineas y quedaría cubierto todos los casos. En odoo 8 no estaba contemplado como otras muchas cosas del POS que aunque estaba integrado en el sistema iba como por libre...

pedrobaeza commented:
Ángel, cómo vas con esto? Algún progreso?

angelmoya commented:
Pensaba ponerme ahora en Agosto, pero la primera quincena ha sido
complicada, tengo intención de aldarle un repaso antes de Septiembre.

El jue., 16 ago. 2018 8:08, Pedro M. Baeza <notifications@github.com>
escribió:

> Ángel, cómo vas con esto? Algún progreso?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/OCA/l10n-spain/pull/862#issuecomment-413436265>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AGcAhV9WG07dUCWgk74gfjKdoEsjBkYYks5uRQxDgaJpZM4UiVG_>
> .
>

pedrobaeza commented:
Por cierto, @Joaco1980, haciéndolo por apuntes no se quedan ni mucho menos fuera la actividad del POS, ya que también generan apuntes. Eso sí, no es un caso de uso que esté en nuestro foco ahora mismo (por lo menos en el mío, y supongo que el de Ángel tampoco), así que deberás revisar y si acaso complementar lo que se haga para que esté funcionando 100% con el POS.

Joaco1980 commented:
@pedrobaeza si eso es precisamente lo que digo, que si se hace por apuntes las operaciones del POS estarian contempladas y el modulo abarcaria mas focos de uso. Antes el modulo tiraba de las facturas y dejaba fuera las simplificadas.

pedrobaeza commented:
OK, entonces mejor estar de acuerdo :)

HaraldPanten commented:
Buenos días @angelmoya 

Has podido hacer algún avance?

angelmoya commented:
He estado mirando el tema de sacar la información desde account mover lines
y he cambiado algunos "for - if" por filtered, pero no he probado con datos
por si se me escapa algo... Esta semana subo lo que tengo

El mié., 5 sept. 2018 10:27, Harald Panten <notifications@github.com>
escribió:

> Buenos días @angelmoya <https://github.com/angelmoya>
>
> Has podido hacer algún avance?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/OCA/l10n-spain/pull/862#issuecomment-418644755>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AGcAhboIzqYBfCYUHzuA2YUkfXwAAueyks5uX4sLgaJpZM4UiVG_>
> .
>

angelmoya commented:
Buenas,

al plantear la busqueda de los importes por apuntes, en lugar de por facturas, me encuentro un par de dudas que quería comentar con todos (Copiaré este mensaje en la lista de correo)

Se había planteado que si se cambiaba el 347 para buscar los importes por apuntes el sistema iría mas rápido e incluso si se hace una importación buena de contabilidad se podría obtener el informe incluso sin tener las facturas.

Actualmente al calcular el 347, para cada cliente se crea un objeto 'l10n.es.aeat.mod347.partner_record', y relacionado con este un listado de objetos 'l10n.es.aeat.mod347.invoice_record' en el que tenemos cada factura, con su tipo de factura y su fecha y su importe, que se calcula como la base imponible y la suma de los impuestos (menos los que sean IRPF).

Si hacemos por apuntes:
- Tengo la fecha contable, pero no la fecha de la factura, actualmente lo estoy haciendo por fecha de factura.
- El tipo de factura me sirve para diferenciar facturas de clientes y de proveedores, esto lo puedo hacer con el tipo de diario de venta o de compra, pero también me sirve para diferenciar si es rectificativa o no. Para hacer esto por apuntes tendría que ver en los apuntes por tipo de cuenta si va al debe o al haber.
- Si queremos mostrar un informe con detalle de las facturas sobre las que hemos realizado el cálculo, pero calcularlo por apuntes pondríamos la referencia del asiento (que sería la factura), pero igualmente necesitaríamos calcular el importe de cada asiento.

Resumiendo, si queremos sólo el importe total de cada partner si que veo que se podría hacer por apuntes, pero también se podría agilizar el calculo que se hace ahora, pero como se quiere el detalle y se esta almacenando en cada factura el importe con un campo calculado y almacenado no veo de que forma podríamos tener la misma funcionalidad pero por apuntes.

Si alquién quiere o puede echarme una mano podemos organizar una conferencia y ver estos detalles técnicos, gracias.

fgarcia-humanoide commented:
Yo no puedo ayudar, solo indicar que el 347 es importante, que no hay 347 desde las versiones del 2015, y que los que hacen facturas con el tpv son un un ipedimento para el progreso de la humanidad.

pedrobaeza commented:
> solo indicar que el 347 es importante, que no hay 347 desde las versiones del 2015

Sí que los hay, y están como PRs, pero obviamente no es la mejor implementación y por eso estamos haciendo esto.

pedrobaeza commented:
> Actualmente al calcular el 347, para cada cliente se crea un objeto 'l10n.es.aeat.mod347.partner_record', y relacionado con este un listado de objetos 'l10n.es.aeat.mod347.invoice_record' en el que tenemos cada factura, con su tipo de factura y su fecha y su importe, que se calcula como la base imponible y la suma de los impuestos (menos los que sean IRPF).

¿Esos registros son necesarios para la AEAT? Si no, pues se pueden deshechar los mismos y que el detalle sea un listado de los asientos.

> Si hacemos por apuntes:
>    Tengo la fecha contable, pero no la fecha de la factura, actualmente lo estoy haciendo por fecha de factura.

Cada asiento guarda una relación con la facturas. Sería tener fecha de factura o fecha del asiento.

> El tipo de factura me sirve para diferenciar facturas de clientes y de proveedores, esto lo puedo hacer con el tipo de diario de venta o de compra, pero también me sirve para diferenciar si es rectificativa o no. Para hacer esto por apuntes tendría que ver en los apuntes por tipo de cuenta si va al debe o al haber.

Para eso está el módulo `account_tax_balance`, del que depende ya `l10n_es_aeat`, y que nos ofrece el tipo de asiento que es:

https://github.com/OCA/account-financial-reporting/blob/9d5d5e7b052fb3e47a0e2a0198902074539d487d/account_tax_balance/models/account_move.py#L10-L18

y precisamente es la técnica utilizada en todos sitios para el 303, 390, 349...

> Si queremos mostrar un informe con detalle de las facturas sobre las que hemos realizado el cálculo, pero calcularlo por apuntes pondríamos la referencia del asiento (que sería la factura), pero igualmente necesitaríamos calcular el importe de cada asiento.

Igualmente, es coger el número de factura si presente, o la referencia del asiento (más que el nº, ya que la referencia es la que puede contener el nº de factura, y así debe importarse en caso de querer de esta manera).

angelmoya commented:
OK @pedrobaeza , una última duda es si sería correcto poner el importe del asiento, o habría que tener algo mas en cuenta.

Por ejemplo para una factura con IRPF:


Cuenta | Empresa | Descripción | Debe | Haber
-- | -- | -- | -- | --
430000 Clientes (euros) | Agrolait | / | 28,00 € | 0,00 €
477000 Hacienda Pública. IVA repercutido | Agrolait | IVA 21% (Bienes) | 0,00 € | 5,25 €
473000 Hacienda Pública, retenciones y pagos a cuenta | Agrolait | Retenciones a cuenta IRPF 9% | 2,25 € | 0,00 €
700000 Ventas de mercaderías en España | Agrolait | [C-Case] Carcasa de computadora | 0,00 € | 25,00 €
  |   |   | 30,25 | 30,25

En este caso yo podría utilizar el importe del asiento que es 30,25€, si no tuviese IRPF también sería correcto utilizar el importe total del asiento.

¿Hay algún caso en el que no me coincida el importe del asiento con lo que tengo que declarar?

En la factura se filtra a piñon para utilizar la base imponible de la factura mas los impuestos que no sean de IRPF.

Joaco1980 commented:
Buenas tardes,

El requisito de Hacienda es tan solo la cifra total en las operaciones, desglosadas por trimestres.

https://www.agenciatributaria.es/static_files/AEAT/Contenidos_Comunes/La_Agencia_Tributaria/Ayuda/Disenyos_de_registro/Ayudas/DR_300_a_399/ficheros/347.pdf

Joaco1980 commented:
@fgarcia-humanoide pues date una vuelta por Zara, Carrefour, Corte Ingles etc etc o cualquier tienda comercial y fijate que te dan si Facturas o Facturas Simplificadas. El verdadero impedimento es la gente que no ve mas alla de lo que le rodea.... pero bueno que aqui estamos para debatir como hacer un buen modulo del modelo 347...

fgarcia-humanoide commented:
Pues si , mejor hacer un odoo completito para que lo pueda usar Zara y Carrefour, que son los típicos usuarios de odoo.

pedrobaeza commented:
> una última duda es si sería correcto poner el importe del asiento, o habría que tener algo mas en cuenta.

@angelmoya la técnica vuelve a ser la misma que con el resto de declaraciones AEAT: mapeo de impuestos que especifican la cantidad que queremos, ya que no nos interesan ni intracomunitarios, ni IRPF, ni esas cosas, por lo que en este caso, será utilizar la base imponible de los impuestos habituales (4%, 10% y 21% de bienes y servicios). Que en un momento dado tenemos que poner o quitar la base de un impuesto que interese, pues ya no se toca código, se tocan datos. ¿Entiendes? 

Echa un vistazo al mapeo de impuestos del 303 por ejemplo, y crea tú un mapeo para el 347 de forma similar. Aunque en este caso no tengamos nº de casillas, puedes dar de alta una línea con casilla por ejemplo 0 que sea la definición que queremos, y tú de manera manual obtienes el resto de datos sobre las move_lines que te obtenga el propio motor.

Joaco1980 commented:
Ojo que los valores a cumplimentar son totales de la operación, no es necesario indicar el tipo de Impuesto aplicado...

HaraldPanten commented:
Buenos días @angelmoya ,

Habéis podido avanzar algo con este tema? Necesitáis ayuda?

Un saludo!

angelmoya commented:
Gracias @HaraldPanten , lo tengo en la lista de TO DO... como no hacía falta hasta final de año lo he dejado pendiente, ya empieza a ser necesario.

Me pongo con eso esta semana y aviso por aquí... la ayuda buena será la review del código.

HaraldPanten commented:
Perfecto @angelmoya ,

Nuestro equipo mirará de irle dando caña con las reviews de código y las pruebas funcionales.

Un saludo!

angelmoya commented:
Buenas, le he dado un repaso y sigo sin ver claro utilizar lo mismo mecanismo que en 303, por que al final lo que necesito es el listado de facturas y los importes brutos, y hacerlo por empresa. Si sólo quisiesemos el importe lo vería bien, pero al final lo que nos da un poco de valor es poder enviar al cliente el listado de facturas para cruzar.

Por lo que entiendo el mecanismo me da apuntes relacionados con los impuestos, si una factura tuviese varios tipos de impuestos tendría varios apuntes relacionados.  Por ejemplo en el 349, si tengo distintos tipos de impuestos en adquisiciones intracomunitarias, el importe total está bien, pero me aparece la factura dos veces en el detalle, porque me parecerá una línea por cada impuesto. Esto si se le quiere enviar a otra empresa para cuadrar será un poco lío que no vean el importe total de la factura en una línea o que vean la factura dos veces.

Otra cosa a tener en cuenta es que con el TVGI Online el 347 va a cambiar mucho.

Voy a revisar algunos errores de instalación, el informe y el envío por email, pero si queréis modificar el funcionamiento para que funcione por apuntes lo tendríamos que mirar, por mi parte lo dejaría así hasta que se tenga claro como se tiene que realizar en el futuro.

Esto creo que sería mejor hablarlo, si queréis organizamos una conferencia los interesados y lo vemos.

pedrobaeza commented:
Y dale, Ángel... no es el mismo que el 303 (aunque parecido), si no el del 349. Es perfectamente posible y viable. Es que no hay que poner los impuestos por separado. Va todo junto. Está bien, déjalo. Ya lo haremos nosotros.

angelmoya commented:
Pedro, eso mismo te estoy diciendo, que es parecido al 349, pero como digo si va por apuntes y tenemos facturas con varios impuestos nos devolverá dos apuntes, y esto para enviar al cliente no me parece muy operativo. Y si tenemos que hacer todo el proceso para arreglar lo que nos devuelve y presentarlo no lo veo operativo, y mas si van a cambiar la forma de presentarlo.

Corrígeme si me equivoco, si tienes una factura con impuestos al 21 y al 10 saldrán dos apuntes.

¿Que hariamos un campo computed, que saque move_line.ids.mapped('invoice_id')?

No te digo que no sea posible y viable, solo que no estoy seguro de verlo claro.

Tendríamos que tener entonces un listado de apuntes de los que saca la información, y de este listado sacar con el mapped que te comento antes las facturas que al final es lo que se va a enviar.

pedrobaeza commented:
No hace falta un campo computed, si no directamente almacenar lo que nos interese, y en este caso es la información agrupada por factura. Para optimizar rendimiento aún más, podrías considerar utilizar un `read_group` (en breve lo voy a añadir yo a los métodos genéricos de `l10n_es_aeat`).

angelmoya commented:
@pedrobaeza , para que lo tengais en cuenta y no duplicar trabajo, estamos haciendolo como decias con mapeo de impuestos y read_group.

pedrobaeza commented:
Genial, porque ya va quedando menos para necesitarlo... Recuerda luego agrupar por `move_id` para obtener el "total" de la factura, en lugar de por `invoice_id`, ya que así funcionará también para asientos importados.

angelmoya commented:
ok, gracias, entre hoy y mañana espero tenerlo

angelmoya commented:
Acabo de hacer un commit pero está en progreso, queda por hacer:

- Poner los impuestos correctos en el mapeo, y ver si es correcto hacer una linea para la clave A y otra linea para la clave B
- Ver la mejor forma de controlar los límites para añadir o no el registro.
- Ver en que casos se añadiría la clave de operación C y si es posible controlar el resto de claves, que en la versión anterior sólo se crean claves A y B.
- Añadir los registros de caja a la clave B del partner correspondiente.
- Limpiar código obsoleto

¿Sería posible recibir dinero de un cliente sin tener una factura? En la versión actual no se contempla ese caso y no se añade el registro tipo C

elarapuntsistemes commented:
@angelmoya 
La normativa fiscal que recoge el carácter obligatorio de la remisión de facturas:

- Artículo 29.2e) de la Ley 58/2003, de 17 de diciembre, General Tributaria (BOE de 18.12.2003)
- Artículo 164, apartado Uno, número 3º de la Ley 37/1992, de 28 de diciembre, del Impuesto sobre el Valor Añadido (BOE de 29.12.1992)
- Artículo 2.1 del Reglamento por el que se regulan las obligaciones de facturación aprobado por el artículo primero del Real Decreto 1619/2012, de 30 de noviembre (BOE de 01.12.2012)
La Agencia Tributaria indica:

“Los empresarios y profesionales están obligados a expedir factura y copia de esta por las entregas de bienes y prestaciones de servicios que realicen en desarrollo de su actividad y a conservar copia de la misma. También deben expedir factura en los supuestos de pagos anticipados, excepto en las entregas intracomunitarias de bienes exentas”

angelmoya commented:
Ok, pero hay un supuesto que sea una factura del año pasado pagada este año.

En ese caso el informe actual no lo contempla, al no haber facturas este año no crea el registro.

fgarcia-humanoide commented:
Los pagos no importan , solo importa la emisión de la factura, en mi humilde opinion.

angelmoya commented:
@fgarcia-humanoide lo que dice el modelo es que hay que indicar lo facturado y lo cobrado en metálico cada año

fgarcia-humanoide commented:
Lo cobrado en metálico que se meta a mano. No se cobra nada en metálico en el caso general.

angelmoya commented:
@fgarcia-humanoide el modelo ya contempla los cobros en efectivo, ya que los toma de los diarios de tipo "cash"

fgarcia-humanoide commented:
I have the same issue in an odoo 11 ubuntu 18.04

File "b'/opt/odoo/odoo-server/odoo/import_xml.rng'", line 0
lxml.etree.XMLSyntaxError: <no detail available>

Can not install modules
Can not update modules

Two days ago installed 120 modules.

Now i can't

etobella commented:
@angelmoya ¿En que estado está este PR? ¿Podemos ayudarte en algo?

angelmoya commented:
Lo último que deje a medio fue por una pregunta que nadie me supo responder. Actualmente en el informe que hay, en el caso de que haga facturas en 2017 y se paguen en efectivo en 2018, si no he hecho facturas en 2018 no aparecerá en el informe, creo que esto no es correcto pero no he tenido confirmación.

Retomaré esto, pero a las malas la versión que hice sin la mejora de funcionar por apuntes contables funcionaba bien. Lo subiré en otra rama por si alguien lo necesita.

Joaco1980 commented:
Desde mi humilde opinion creo que solo hay que indicar lo cobrado en metalico del año que se presenta. En el supuesto de que se facture en 2017 y se cobre en 2018 algo en metalico el importe de operaciones de facturacion irian en 2017 y lo cobrado en metalico en la declaracion de 2018.
Por lo tanto hay que buscar en el diario del tipo “cash” lo cobrado a un determinado proveedor filtrado por trimestres. Si puedes colgarlo para poder teestarlo te lo agradeceria.

Saludos

Un saludo

pedrobaeza commented:
A ver, que ese caso tan extremo bloquee esto me parece un poco absurdo. Ángel, si no puedes/quieres terminarlo, no pasa nada. Dilo y ya está. Pero no nos marees...

angelmoya commented:
No mareo, hace un mes que dije como iba el tema y un caso extraño no contemplado en la versión anterior, y no tuve feedback y lo deje aparcado.

Ya publique una versión funcional a falta de la parte de cambiar los asientos que es lo que tengo pendiente.

En mi opinión se debería haber fusionado sin esa mejora, y hacer otro pr para esta parte, pero en cualquier caso continuaré con eso.

Si queréis revisar hace 28 días puse un mensaje diciendo cómo estaba y lo que quedaba, si alguien quiere empezar a revisar, continuar el trabajo o hacer alguno de los puntos pendientes adelante.

angelmoya commented:
Si alguien quiere probarlo hay una tarea sencilla que se puede hacer, que es preparar los mapeos de impuestos, que no están creados. Con esto se podría probar y solo faltaría controlar los límites.

pedrobaeza commented:
Pero Ángel, si es que la principal mejora sigue siendo ésa, la de hacerlo por apuntes, y es lo que no has hecho. Por lo demás, hay desde hace 2 años PRs funcionales que se aplican igual a v11. De hecho, creé una rama para la v11 en: https://github.com/Tecnativa/l10n-spain/tree/11.0-l10n_es_aeat_mod347, pero no está propuesta porque supuestamente llegaba esta mejora.

¿Entonces lo de los apuntes está o no está hecho?

angelmoya commented:
A falta de añadir el mapeo, tiene uno de pruebas con un par de codigos, y los límites, para que no muestre si suma menos de 3000.

angelmoya commented:
En esta rama de añadió desde junio también el proceso de envío por email

etobella commented on l10n_es_aeat_mod347/models/mod347.py:
```diff
@@ -0,0 +1,852 @@
+
+    def _create_partner_records(self, key, map):
+        map_a = self.env.ref(map)
+        taxes_a = self._get_taxes(map_a)
```
¿Por que usáis los IVAs? De hecho, no es relevante que IVA se esten usando. Nosatros no tenemos IVA (empresa de servicios médicos) y debemos presentar igual el 347 con los clientes que superen dicho importe. En mi opinión debería incluirse todos los receivables.

pedrobaeza replied:
Se usa los impuestos porque el importe de los 3000 es sin tener en cuenta el IRPF. Entonces todo eso se necesita saber. En tu caso, si te has creado nuevos impuestos, tendrás que mapearlos como válidos para el 347.

etobella commented on l10n_es_aeat_mod347/models/mod347.py:
```diff
@@ -0,0 +1,852 @@
+                  u'límite (1)'),
+            ('B',
+             u'B - Entregas de bienes y servicios superiores al límite (1)'),
+            ('C',
```
La clave C es para cuando cobramos a nombre de un tercero. En principio diria que no hay módulos en la OCA para esto. Por lo que no debería preocuparnos.

etobella reviewed (changes requested):
Algunos comentarios sobre lo que he podido analizar.
Creo que @Joaco1980 tiene razón sobre como debería imputarse la parte del efectivo.

Joaco1980 commented:
Yo creo que las que lleven retencion IRPF quedan fuera del modelo ya que se presentan en otro modelo diferente 111 trimestralmente y 190. Si lo puede confirmar alguien mejor que mejor.

> El 22 ene 2019, a las 18:14, Pedro M. Baeza <notifications@github.com> escribió:
> 
> @pedrobaeza commented on this pull request.
> 
> In l10n_es_aeat_mod347/models/mod347.py:
> 
> > +    @api.model
> +    def _get_taxes(self, map):
> +        tax_obj = self.env['account.tax']
> +        # Obtain all the taxes to be considered
> +        tax_templates = map.mapped('tax_ids').mapped('description')
> +        if not tax_templates:
> +            raise exceptions.Warning(_('No Tax Mapping was found'))
> +        # search the account.tax referred to by the template
> +        taxes = tax_obj.search(
> +            [('description', 'in', tax_templates),
> +             ('company_id', 'child_of', self.company_id.id)])
> +        return taxes
> +
> +    def _create_partner_records(self, key, map):
> +        map_a = self.env.ref(map)
> +        taxes_a = self._get_taxes(map_a)
> Se usa los impuestos porque el importe de los 3000 es sin tener en cuenta el IRPF. Entonces todo eso se necesita saber. En tu caso, si te has creado nuevos impuestos, tendrás que mapearlos como válidos para el 347.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

fgarcia-humanoide commented:
Lo que se haya aportado en otros modelos, no se declara. Por ejemplo:

Facturas con retencion ya se declaran en el 190 y 180. 
Facturas intracomunitarias ya se declaran en el 349.

No encuentro el soporte legislativo de esto.

oca-travis committed:
[UPD] Update l10n_es_aeat_mod347.pot

javimurcia commented:
Como dice @fgarcia-humanoide no se declaran ni importaciones o exportaciones, o cualquier otra importación para la que haya un suministro de información por otro medio (facturas con retención o intracomunitarias). Exactamente el boe dice:

g) Las importaciones y exportaciones de mercancías (...)

i) En general, todas aquellas operaciones respecto de las que exista una obligación periódica de suministro de información a la Administración tributaria estatal y que como consecuencia de ello hayan sido incluidas en declaraciones específicas diferentes a la regulada en esta subsección y cuyo contenido sea coincidente.

El artículo 33 en la sección 2 del RD 1065/2007 tiene todas las excepciones al 347

yajo commented on l10n_es_aeat_mod347/wizard/calculate_mod347_records.py:
```diff
@@ -0,0 +1,371 @@
+                    'calculation_date': time.strftime('%Y-%m-%d %H:%M:%S')
+                })
+
+        except Exception, ex:
```
Esto no vale en Python 3:

```
*** Error compiling '/opt/odoo/custom/src/l10n-spain/l10n_es_aeat_mod347/wizard/calculate_mod347_records.py'...
  File "/opt/odoo/custom/src/l10n-spain/l10n_es_aeat_mod347/wizard/calculate_mod347_records.py", line 352
    except Exception, ex:
                    ^
SyntaxError: invalid syntax
```

angelmoya replied:
Ese fichero es código obsoleto, ya lo he borrado, no llegaba a ejecutarse

mtbochaca committed:
Translated using Weblate (Catalan)

Currently translated at 81.8% (166 of 203 strings)

Translation: l10n-spain-9.0/l10n-spain-9.0-l10n_es_aeat_mod347
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-9-0/l10n-spain-9-0-l10n_es_aeat_mod347/ca/

angelmoya commented:
Ya esta funcionando también con el filtro del limite y he borrado el código obsoleto.

He creado un mapeo de impuestos, pero si alguien puede revisar si falta o sobra alguno que me avise.

Mientras escribía me he dado cuenta que falta:
- el calculo de los importes de los trimestres.
- hacer que una factura o asiento no aparezca, se puede por partner, no se si esto es necesario.

angelmoya commented:
Ya esta lo que faltaba de los importes de trimestres y poder marcar empresas, facturas o movimientos para que no se tengan en cuenta.

pedrobaeza commented:
OK, pero mira por favor el status tanto de runbot como de Travis.

angelmoya commented:
A falta de incrementar coverage, si alguien quiere revisar el resto ya parece estar bien

Joaco1980 commented:
A mi me da el siguiente error al intentar calcular el 2018:

  File "/opt/odoo/addons_link/l10n_es_aeat/models/l10n_es_aeat_report.py", line 273, in button_calculate
    res = self.calculate()
  File "/opt/odoo/addons_custom/l10n_es_aeat_mod347/models/mod347.py", line 315, in calculate
    self._create_cash_moves()
  File "/opt/odoo/addons_custom/l10n_es_aeat_mod347/models/mod347.py", line 301, in _create_cash_moves
    ('type', '=', 'B')
  File "/opt/odoo/odoo/models.py", line 1480, in search
    res = self._search(args, offset=offset, limit=limit, order=order, count=count)
  File "/opt/odoo/odoo/models.py", line 3773, in _search
    query = self._where_calc(args)
  File "/opt/odoo/odoo/models.py", line 3563, in _where_calc
    e = expression.expression(domain, self)
  File "/opt/odoo/odoo/osv/expression.py", line 668, in __init__
    self.parse()
  File "/opt/odoo/odoo/osv/expression.py", line 846, in parse
    raise ValueError("Invalid field %r in leaf %r" % (left, str(leaf)))
ValueError: Invalid field 'type' in leaf "<osv.ExtendedLeaf: ('type', '=', 'B') on l10n_es_aeat_mod347_partner_record (ctx: )>"

He probado en runbot a meter una factura y pagarla en cash y da el mismo error.

Joaco1980 commented on l10n_es_aeat_mod347/wizard/l10n_es_aeat_mod347_send_mail.py:
```diff
@@ -0,0 +1,11 @@
+from openerp import api, models
```
from odoo import api, models

MiquelRForgeFlow replied:
@pedrobaeza falta este cambio :)

Joaco1980 commented:
A los usuarios sin permisos de facturacion les salta un error en res.partner
![787dbf06-b397-43ee-9409-c84d42dfe077](https://user-images.githubusercontent.com/14363121/52305952-c5179100-2996-11e9-8e54-e62a377b1ce2.jpeg)

Si este no es el lugar correcto para poner este tipo de cosas decirlo eh.

pedrobaeza commented:
Voy a retomar yo el desarrollo desde aquí para darle el último empujón, poner algunos tests y demás.

angelmoya committed:
[MIG+IMP] l10n_es_aeat_mod347: Migration to 11.0 + enhancements

pedrobaeza commented:
Histórico de commits completo recuperado.

pedrobaeza commented on l10n_es_aeat_mod347/models/mod347.py:
```diff
@@ -0,0 +1,824 @@
+        # search the account.tax referred to by the template
+        taxes = tax_obj.search(
+            [('description', 'in', tax_templates),
+             ('company_id', 'child_of', self.company_id.id)])
```
@angelmoya alguna razón especial para poner esto aquí?

angelmoya replied:
No entiendo a que te refieres @pedrobaeza

pedrobaeza replied:
OK, entonces no es por nada en especial.

pedrobaeza committed:
[FIX+IMP] l10n_es_aeat_mod347: Migration refinements + enhancements

* Add controller for automatic confirmation/rejection
* Improve data model
* Clean code
* Clean UI
* Spanish translation
* Tests
* ...

pedrobaeza commented:
Ya he terminado por mi parte la mejora del módulo incluyendo algunas cosas más de las hechas por @angelmoya, entre algunas:

* Mejorar modelo de datos, quitando campos o modelos no necesarios para reducir la huella en la BD.
* Limpieza de código.
* Limpieza y refinamiento de UI.
* Traducción a español
* Tests
* Incluir el controlador para poder aceptar/rechazar el email enviado (¡mola mucho, jeje!):
  ![imagen](https://user-images.githubusercontent.com/7165771/52480654-95da6d00-2bac-11e9-8a41-8fa17b6c19e4.png)
* Varias más difíciles de enumerar.

En cuanto a la autoría del módulo, la he reseteado a solo PESOL y Tecnativa, ya que todo el código actual es de los dos y no queda nada de anteriores. Sin embargo, siguen constando en la sección del copyright del manifiesto y en los contribuidores.

Si queréis ir revisando, yo paso a ponerlo en clientes de prueba para detectar algún último problema.

Joaco1980 commented:
Funciona perfecto! Un gran trabajo, la confirmacion online la verdad que le da un plus al modulo.

Yo como apuntes diria 2 cosas que he visto:

- Las facturas rectificativas en el pdf que se le envia a los clientes se podrian marcar o con otro color como en el backend o con un signo “-“ justo delante del importe. 

- Las facturas del tipo A en el detalle y en el informe el numero de factura coge la secuencia interna de odoo cuando creo que seria mas claro mostrar el numero de factura del proveedor.

Salen los arrendamientos y las quw tienen retenciones pero bueno sabiendolo y teniendo la opcion de marcar la casilla de no incluir en 347 no es mayor problema.

pedrobaeza commented:
Buenas, @Joaco1980 me parecen buenas sugerencias las que has hecho. Ya están incluidas.

Sobre:

> Salen los arrendamientos y las quw tienen retenciones pero bueno sabiendolo y teniendo la opcion de marcar la casilla de no incluir en 347 no es mayor problema.

Esto hemos tenido debate @albertgafic y yo y efectivamente una práctica habitual es que aquí solo sale "lo que en otros modelos no sale", pero implementar eso sería:

- Hacer más lenta la búsqueda.
- Generar malentendidos cuando hay facturas mixtas con líneas con retención y sin retención.
- Tener que rehacer más cosas del módulo.

Por eso, no es tampoco del todo problemático que salgan ahí, ya que al final es un informativo, y hubiera hecho más esfuerzo por implementarlo y optimizarlo si no supiera que esto va a ser sustituido en breve por un sistema online tipo SII (TGVI - https://www.agenciatributaria.es/AEAT.internet/Inicio/RSS/Todas_las_Novedades/Le_interesa_conocer/Informacion_sobre_la_proxima_campana_de_declaraciones_informativas_y_recomendaciones_en_la_identificacion.shtml), así que tampoco merece la pena mayor esfuerzo. Ya he añadido el punto a los known issues de todas formas.

Si estás de acuerdo con el módulo, puedes hacer una review aprobando el PR.

Daniel-CA commented:
Yo he tratado de instalarlo en varios entornos y me ha dado este error en una de las vistas:
```
raise ValueError("Wrong value for %s: %r" % (self, value))
odoo.tools.convert.ParseError: "Wrong value for l10n.es.aeat.map.tax.line.field_type: 'both'" while parsing /opt/odoo/v11/temp-links/l10n_es_aeat_mod347/data/tax_code_map_mod347_data.xml:12, near
<record id="aeat_mod347_map_a" model="l10n.es.aeat.map.tax.line">
      <field name="map_parent_id" ref="aeat_mod347_map"/>
      <field name="field_number">01</field>
      <field name="name">A - Adquisiciones de bienes y servicios superiores al límite</field>
      <field name="to_regularize" eval="False"/>
      <field name="move_type">regular</field>
      <field name="field_type">both</field>
      <field name="sum_type">both</field>
      <field name="inverse" eval="False"/>
      <field name="tax_ids" eval="[(6, False, [         ref('l10n_es.account_tax_template_p_iva21_bc'),         ref('l10n_es.account_tax_template_p_iva21_sc'),         ref('l10n_es.account_tax_template_p_iva4_bi'),         ref('l10n_es.account_tax_template_p_iva4_sc'),         ref('l10n_es.account_tax_template_p_iva10_bi'),         ref('l10n_es.account_tax_template_p_iva21_bi'),         ref('l10n_es.account_tax_template_p_iva10_bc'),         ref('l10n_es.account_tax_template_p_iva4_bc'),         ref('l10n_es.account_tax_template_p_iva10_sc'),         ref('l10n_es.account_tax_template_p_iva0_nd'),         ref('l10n_es.account_tax_template_p_iva0_bc'),         ref('l10n_es.account_tax_template_p_iva0_ns'),         ref('l10n_es.account_tax_template_p_iva12_agr'),         ref('l10n_es.account_tax_template_p_req014'),         ref('l10n_es.account_tax_template_p_req05'),         ref('l10n_es.account_tax_template_p_req52'),         ref('l10n_es.account_tax_template_p_iva4_isp_1'),         ref('l10n_es.account_tax_template_p_iva10_isp_1'),         ref('l10n_es.account_tax_template_p_iva21_isp_1'),         ref('l10n_es.account_tax_template_p_iva105_gan'),         ])]"/>
  </record>
```

pedrobaeza commented:
Este PR también contiene cambios en l10n_es_aeat, así que no debes perderlos y debes actualizar dicho módulo.

Daniel-CA commented:
Gracias @pedrobaeza, era eso, no me había fijado que el módulo de la AEAT tenía modificaciones también.

Joaco1980 commented:
Yo ya tengo funcionando el modulo y me estan enviando confirmaciones y hasta ahora todo ok. Pero he detectado un problemilla de usabilidad.

Imaginad que por cualquier motivo tienes que descartar o un apunte o una factura por cualquier motivo con el check de "no incluir en 347". Hay que cancelar el modelo y volver a recalcular todo de nuevo. Si has enviado mails me imagino que si confirman ya no sera valido porque van con tokens. Fuera aparte que si ya tienes confirmaciones las pierdes y deberias volverlas a reconfirmar. 
Seria posible poder borrar en el detalle de operaciones una operacion concreta, o por ejemplo volver a recalcular pero solo un partner?

Como lo veis?

pedrobaeza commented:
Para eso lo que he dejado es que se editen a mano los datos (faltarían los trimestres que ahora lo habilito), pero lo demás es ya un mecanismo más costoso y no tengo pensado hacerlo dado que éste en teoría es el último año que se utilizará el 347 en favor del TGVI.

Joaco1980 commented:
Si ya vi que se podia editar el total, pero los trimestrales irian mal y seguramente saltaría una campana en Hacienda jajaja

Roodin commented:
¿Por qué en el mapeo figuran impuestos intracomunitarios y de exportaciones si no deben considerarse en el 347?

pedrobaeza commented:
@albertgafic dijo de incluirlas.

pedrobaeza commented:
Incluída conversión de los códigos de provincia de estilo matrícula a estilo CP y quitado de solo lectura los trimestres para aquellos que manipulan manualmente el 347 (ejem).

AlbertCabedo commented:
no hay que incluirlo, solo inversión del sujeto pasivo Nacional. un error de selección en el csv, disculpas

pedrobaeza commented:
OK, ya las he excluido. La verdad es que yo ni me fijé en esa parte como Ángel y Albert lo habían acordado ya, pero eso, malentendidos solucionados!

pedroortega-punt commented:
El 347 ya es funcional?

Gracias.

MiquelRForgeFlow commented:
Creo que estaría bueno que se corrigieran algunos errores no bloqueantes detectados por el LINT_CHECK

pedrobaeza commented:
Miquel, no son de este módulo.

pedroortega-punt commented:
@pedrobaeza 
Entonces se puede utilizar?

pedrobaeza commented:
Por qué no lees el hilo, @vipvalen?

pedroortega-punt commented:
@pedrobaeza 
Lo he segido, pero tengo muchas dudas, no veo que se diga firmemente que ya se puede utilizar y no tiene problemas.

pedrobaeza commented:
Se puede utilizar, pero obviamente no puedo asegurar al 100% que no haya problemas. Para eso estáis vosotros los revisores.

Joaco1980 commented:
Yo lo tengo funcionando, he comprobado que todos los registros eran correctos a la vieja usanza, incluye apuntes del POS (por fin!!), he enviado los mails de confirmación a todos los contactos, la respuesta es correcta porque los valida correctamente. Incluso he exportado el archivo y he probado a importarle en aeat y todo ok! Así que por mi parte puedo afirmar que funciona correctamente con la salvedad de lo comentado mas arriba que ya a aclarado Pedro que no merece la pena tocarlo porque este es el ultimo año de este tipo de presentación.


> El 13 feb 2019, a las 11:15, Pedro M. Baeza <notifications@github.com> escribió:
> 
> Se puede utilizar, pero obviamente no puedo asegurar al 100% que no haya problemas. Para eso estáis vosotros los revisores.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

pedrobaeza commented:
Bueno, finalmente he implementado la opción de recalcular para un registro de empresa en concreto desde el formulario de detalle por petición popular de varios clientes con casuísticas de no tener "muy finos" los datos, así que ya se queda completado eso también aunque solo sirva para este año.

ghost commented:
Un gran trabajo. 
Muchas gracias @pedrobaeza

javiermatos commented:
Muchas gracias por el trabajo @pedrobaeza 
¿Cuándo se integrarán los cambios en la rama 11.0? ¿Quedan pendientes pruebas por hacer? Si hubiera algo por probar decir que en nuestra empresa vamos a utilizarlo para la generación del modelo y que podríamos dar feedback de cualquier punto que se estime necesario.

pedrobaeza commented:
@javiermatos podéis usarlo ya aunque esté como PR. Quiero mantenerlo así por si surgen más cambios a hacer.

angelmoya commented:
Sobre lo que decía de usar la fecha contable o la fecha de factura, un cliente me dice que lo correcto es usar la fecha contable, porque si un año no contabilizamos una factura no la tenemos que presentar ese año, aunque no coincida el importe, y se tendría que presentar el año siguiente, que tampoco coincidirá.

Si esto es correcto, ya estaría bien.

Otra cosa a tener en cuenta, en teoría es ilegal confirmar datos de 347, aunque si se puede informar, pero el botón de validar  el importe por parte de la otra empresa en teoría no sería legal... aunque está muy bien poder hacerlo.

pedrobaeza commented:
> Sobre lo que decía de usar la fecha contable o la fecha de factura, un cliente me dice que lo correcto es usar la fecha contable, porque si un año no contabilizamos una factura no la tenemos que presentar ese año, aunque no coincida el importe, y se tendría que presentar el año siguiente, que tampoco coincidirá

Una cosa es la teoría, y otra la práctica en la que la empresa proveedora emite la factura en 2018, y lo declara en su 347, pero tú lo quieres/tienes que poner en 2019, y por tanto cambias la fecha contable a 2019. Ahí tendrías el descuadre.

> Otra cosa a tener en cuenta, en teoría es ilegal confirmar datos de 347, aunque si se puede informar, pero el botón de validar el importe por parte de la otra empresa en teoría no sería legal... aunque está muy bien poder hacerlo.

A ver, no existe ninguna validación real y aprobada por la AEAT, ni por teléfono, ni por mail ni por este botón. Es simplemente para cuadrar los datos y que no haya "sorpresas", así que es igual de válido/no válido. Con la entrada del TGVI será cuando haya una validación real legal.

pedrobaeza commented:
Si todos estáis de acuerdo y lo habéis probado sin más feedback, procedo a fusionar. Yo no he tenido ninguna otra reclamación.

pedrobaeza commented:
Fusiono entonces... Gracias a todos/as por la revisión.
